src/utils/cart.js

```diff
@@ -0,0 +1,37 @@
+const token = localStorage.getItem("UserToken");
+export const addToCart = async (product, cartDispatch, postToCart) => {
+  try {
+    const newCart = await postToCart(product, token);
+    if (newCart) {
+      cartDispatch({ type: "ADD_TO_CART", payload: newCart });
+    }
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const removeFromCart = async (id, cartDispatch, deleteFromCart) => {
+  try {
+    const deletedItem = await deleteFromCart(id, token);
+    if (deletedItem) {
+      cartDispatch({ type: "REMOVE_FROM_CART", payload: deletedItem });
+    }
+  } catch (err) {
```

Comment:
try calling dispatch after api call only, you dont have to do try catch two time then

Reply:
Thanks for that